lib/post_destroyer.rb

```diff
@@ -45,6 +45,14 @@ def initialize(user, post, opts = {})
   end
 
   def destroy
+    payload = WebHook.generate_payload(:post, @post)
+    topic = @post.topic
+
+    if @post.is_first_post? && topic
+      topic_view = TopicView.new(topic.id, Discourse.system_user)
+      topic_payload = WebHook.generate_payload(:topic, topic_view, WebHookTopicViewSerializer)
+    end
```

Comment:
Since `Post` and `Topic` are only soft destroyed, perhaps we can only generate the payload after it has been trashed?

Reply:
After a post is soft deleted by author we changing the raw with `(post withdrawn by author...` text. I think it's good to keep it like all other "destroyed" webhooks.